src/Chronicle.Integrations.EFCore/Persistence/EFCoreSagaState.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Chronicle.Integrations.EFCore.Repositories;
+
+namespace Chronicle.Integrations.EFCore.Persistence
+{
+    internal class EFCoreSagaState : ISagaStateRepository
+    {
+        public ISagaStateDBRepository SagaStateDBRepository { get; }
```

Comment:
To me, it makes no sense to introduce another abstraction in the component beeing already internal. I'd completely remove the 'ISagaStateDBRepository' and use the Context object directly.